Source/Core/Core/AchievementManager.cpp

```diff
@@ -729,6 +887,22 @@ AchievementManager::RichPresence AchievementManager::GetRichPresence()
   return rich_presence;
 }
 
+void AchievementManager::DisableRestoreAchievementManager(bool disable)
+{
+  m_disabled = disable;
+  if (m_disabled)
+  {
+    INFO_LOG_FMT(ACHIEVEMENTS, "Achievement Manager has been disabled.");
+    OSD::AddMessage("Please close all games to re-enable achievements.", OSD::Duration::VERY_LONG,
+                    OSD::Color::RED);
+    CloseGame();
+  }
```

Comment:
You probably want the enabled -> disabled & closed state transition to happen under the lock so that you can't run into a situation where a delayed call to the volume hashing re-sets `m_game_hash` after you have set `m_disabled = true`.